Sources/Purchasing/Configuration.swift

```diff
@@ -184,3 +186,36 @@ import Foundation
     }
 
 }
+
+// MARK: - API Key Validation
+
+// Visible for testing
+extension Configuration {
+
+    enum APIKeyValidationResult {
+        case platformSpecific
+        case invalidPlatform
+        case legacy
+    }
+
+    static func validate(apiKey: String) -> APIKeyValidationResult {
+        if apiKey.hasPrefix(Self.appleKeyPrefix) {
+            return .platformSpecific
+        } else if apiKey.contains("_") {
```

Comment:
This logic is very simple, but I think adding a regex is unnecessary.

Reply:
totally agree, this is good enough. 
It might be confusing for maintainers who don't have enough context, though. 
Maybe we could add some clarification? 
like, adding a comment specifying that: 
```
legacy key format:       "CtDdmbdWBySmqJeeQUTyrNxETUVkajsJ"
apple key format:        "apple_CtDdmbdWBySmqJeeQUTyrNxETUVkajsJ"
other platforms format:  "otherplatform_CtDdmbdWBySmqJeeQUTyrNxETUVkajsJ"
```

Reply:
Added :=1